metrics/amplitude.js

```diff
@@ -193,10 +191,33 @@ module.exports = {
           region: data.region,
           os_name: data.os,
           os_version: data.osVersion,
-          device_model: data.formFactor,
-          event_properties: mapEventProperties(eventType, eventGroup, eventCategory, eventTarget, data),
-          user_properties: mapUserProperties(eventGroup, eventCategory, data)
-        });
+          device_model: data.formFactor
+        };
+
+        const events = [
+          pruneUnsetValues({
+            ...eventData,
+            event_type: `${eventGroup} - ${eventType}`,
+            time: event.time,
+            event_properties: mapEventProperties(eventType, eventGroup, eventCategory, eventTarget, data),
+            user_properties: mapUserProperties(eventGroup, eventCategory, data)
+          })
+        ];
+
+        if (mapping.after) {
```

Comment:
Might be worthwhile to add a comment here describing when/why `mapping.after` might be needed, or exactly what is does.